src/core/definePage.spec.ts

```diff
@@ -254,4 +254,128 @@ export default {
       path: '/custom',
     })
   })
+
+  describe('error handling', () => {
+    it('handles syntax errors gracefully when extracting definePage', async () => {
+      const codeWithSyntaxError = `
+<script setup>
+definePage({
+  name: 'test',,  // syntax error: extra comma
+  path: '/test'
+})
+</script>
+
+<template>
+  <div>hello</div>
+</template>
+      `
+      
+      const result = await definePageTransform({
+        code: codeWithSyntaxError,
+        id: 'src/pages/broken.vue?definePage&vue',
+      })
+
+      // Should return empty object instead of crashing
+      expect(result).toBe('export default {}')
+    })
+
+    it('handles syntax errors gracefully when removing definePage from source', async () => {
+      const codeWithSyntaxError = `
+<script setup>
+const a = 1
+definePage({
+  name: 'test',,  // syntax error: extra comma
+  path: '/test'
+})
+const b = 1
+</script>
+
+<template>
+  <div>hello</div>
+</template>
+      `
+      
+      const result = await definePageTransform({
+        code: codeWithSyntaxError,
+        id: 'src/pages/broken.vue',
+      })
+
+      // Should return undefined (no transform) instead of crashing
+      expect(result).toBeUndefined()
+    })
+
+    it('handles malformed definePage object gracefully', async () => {
```

Comment:
This test is redundant, syntax errors are already tested. Ensure you don't have duplicated testss